snippet_parser/core.py

```diff
@@ -113,6 +117,100 @@ def _fast_parse(self, wikitext):
         except mwparserfromhell.parser.ParserError:
             return None
 
+    def extract_from_sentences(self, wikitext, sentences):
+        for sent in sentences:
+            start = wikitext.find(sent)
+            marked = _SENTENCE_MARKER_MARKUP.format(sent = sent)
+            if start >= 0:
+                end = start + len(sent)
+                wikitext = wikitext[:start] + marked + wikitext[end:]
+            else:
+                print("Not found: ", sent)
```

Comment:
We don't really expect this to happen, right? That would indicate a bug in CD?

If so, I think for now we can just raise a ValueError here. Printing like this is nice for debugging but when this is actually deployed we will not notice it (because nobody is looking at the logs).

Reply:
Yes, that means there’s a bug in CD, so CH can’t find the sentence in the wikitext, but the occurrence is only 24 among 37487 sentences. I’ll inspect it further and raise a ValueError here for now.